artifact/azure/signer.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2025 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/mendersoftware/mender-artifact/artifact"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azkeys"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/minio/sha256-simd"
+	"github.com/pkg/errors"
+)
+
+type azureClient interface {
+	GetKey(ctx context.Context, name string, version string,
+		options *azkeys.GetKeyOptions) (azkeys.GetKeyResponse, error)
+	Sign(ctx context.Context, name string, version string, parameters azkeys.SignParameters,
+		options *azkeys.SignOptions) (azkeys.SignResponse, error)
+}
+
+type azureKeyVault struct {
+	keyName    string
+	keyVersion string
+	client     azureClient
+}
+
+func NewKeyVaultSigner(keyName string) (*azureKeyVault, error) {
+	keyVaultName := os.Getenv("KEY_VAULT_NAME")
+	if keyVaultName == "" {
+		return nil, fmt.Errorf("azure: no key vault name specified")
+	}
```

Comment:
Some input sensitization here would be great to prevent bad URL inputs.
The vault name needs to be alphanumeric + hyphen.
https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftkeyvault